vnext/local-cli/config/dependencyConfig.js

```diff
@@ -0,0 +1,104 @@
+const fs = require('fs')
+const path = require('path');
+const glob = require('glob');
+const xmldoc = require('xmldoc');
+
+function dependencyConfigWindows(folder, userConfig = {}) {
+  const sourceDir = userConfig.sourceDir || findWindowsAppFolder(folder);
+
+  if (!sourceDir) {
+    return null;
+  }
+
+  var packageName = null;
+  const packageIDL = findPackageProviderIDL(sourceDir);
+  if (packageIDL){
+    packageName = parsePackageIDLFile(packageIDL);
+  }
+  var cppProjFile = null;
+  var csProjectFile = null;
+  if (packageName)
+  {
+    cppProjFile = findCppProject(sourceDir, packageName);
+    csProjectFile = findCSProject(sourceDir,packageName);
+  }
+
+  var projGUID = null;
+  if (cppProjFile)
+  {
+    const proj = readProject(cppProjFile);
+    var groupNode = proj.childNamed('PropertyGroup');
+    if (groupNode){
+      var nameNode = groupNode.childNamed('ProjectGuid');
+      if (nameNode){
+        projGUID = nameNode.val;
+      }
+    }
+  }
+
+  return {
+    sourceDir,
+    packageIDL,
+    packageName,
+    cppProjFile,
+    csProjectFile,
+    projGUID,
+  }
+}
+
+function findWindowsAppFolder(folder) {
+  const winDir = 'windows';
+  const joinedDir = path.join(folder, winDir);
+  if (fs.existsSync(joinedDir)) {
+    return joinedDir;
+  }
+
+  return null;
+}
+
+// assumption is every cpp native module will have a ReactPackageProvider.idl defined
+function findPackageProviderIDL(folder) {
+  const PackageIDLPath = glob.sync(path.join('**', 'ReactPackageProvider.idl'), {
```

Comment:
It looks like we are being pretty specific about some other file locations, and chance we can do the same with this one?  I'd like to avoid the glob call during config if possible.  -- It can be pretty expensive if there are a lot of files in the package. -- And the config call happens on every build, and would add up with lots of community module dependencies.

And if we do end up doing the glob, can we make sure to ignore the files that the android / ios ones ignore -- in particular: ` 'node_modules/**', '**/build/**', '**/debug/**', 'Examples/**', 'examples/**'`

It does look like the android CLI at least does also do some glob calls.  I wonder if we should also add to their glob ignores to ignore the locations we are ignoring?  (I assume we are ignoring those directories because they have so many files and make things slow?)

Reply:
We first check if there is a "windows" directory under the specific native module directory, this directory is passed in to this function: "folder".

So we only calling glob with cwd parameter set to this windows folder(c:\rnw\node_modules\react-native-video\windows, i.e.), which we know exactly what directories to avoid. In the ignore parameter, I set Debug, Release and "Generated Files", they are three build directories which can contain many files.

Same, android does not do glob globally, it is run under directory android and should not be problem too.

---
In reply to: [389067492](https://github.com/microsoft/react-native-windows/pull/4251#discussion_r389067492) [](ancestors = 389067492)